src/rust/engine/process_execution/src/lib.rs

```diff
@@ -275,6 +276,51 @@ impl InputDigests {
     })
   }
 
+  pub async fn new_from_merged(store: &Store, from: Vec<InputDigests>) -> Result<Self, String> {
+    Ok(Self {
+      complete: store
+        .merge(
+          from
+            .iter()
+            .map(|input_digests| input_digests.complete.clone())
+            .collect(),
+        )
+        .await?,
+      nailgun: store
+        .merge(
+          from
+            .iter()
+            .map(|input_digests| input_digests.nailgun.clone())
+            .collect(),
+        )
+        .await?,
+      input_files: store
+        .merge(
+          from
+            .iter()
+            .map(|input_digests| input_digests.input_files.clone())
+            .collect(),
+        )
+        .await?,
+      immutable_inputs: Itertools::concat(
+        from
+          .iter()
+          .map(|input_digests| input_digests.immutable_inputs.clone()),
+      )
+      .into_iter()
+      .unique()
+      .collect(),
+      use_nailgun: Itertools::concat(
```

Comment:
Hm... it occurs to me that `use_nailgun` should probably be a `BTreeSet` rather than a `Vec`, so that it is sorted. Could then drop the `unique` here.

Reply:
Will add to backlog

Reply:
https://github.com/pantsbuild/pants/issues/15586